h/static/scripts/tests/controllers/character-limit-controller-test.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+var CharacterLimitController = require('../../controllers/character-limit-controller');
+var util = require('./util');
+
+describe('CharacterLimitController', function () {
+
+  /**
+   * Make a <textarea> with the character limit controller enhancement
+   * applied and return the various parts of the component.
+   */
+  function component(value, maxlength) {
+    maxlength = maxlength || 250;
+
+    var template = '<textarea class="js-character-limit" data-maxlength="';
+    template += maxlength + '">';
+    if (value) {
+      template += value;
+    }
+    template += '</textarea><span class="js-character-counter">Foo</span>';
+
+    var containerEl = util.setupComponent(document, template, {
+      '.js-character-limit': CharacterLimitController,
+    });
+    var counterEl = containerEl.querySelector('.js-character-counter');
+    var textarea = containerEl.querySelector('.js-character-limit');
+    var ctrl = textarea.controllers[0];
+
+    return {
+      containerEl: containerEl,
+      counterEl: counterEl,
+      textarea: textarea,
+      ctrl: ctrl,
+    };
+  }
```

Comment:
The problem is that if you've used `beforeEach()` to initialize text fixtures in your tests and you can get away with doing that because so far all the tests need the same fixtures initialized in the same way, then I come along later to edit a test or add a new one to that suite, and my new test needs the test fixtures initialized differently, then I can't add the test that I want to, not without refactoring first.

`beforeEach()` is fine for any code that you absolutely know that you want to execute before every test in the suite, including future tests not written yet. Something like resetting a database for example. But I'd say that creating test fixture objects that are actually going to be used in the tests never comes under this.